crates/nu-cli/src/commands/commandline/edit.rs

```diff
@@ -61,6 +67,13 @@ impl Command for CommandlineEdit {
             repl.buffer = str;
             repl.cursor_pos = repl.buffer.len();
         }
+
+        if call.has_flag(engine_state, stack, "accept")? {
+            engine_state
+                .immediately_accept
+                .store(true, Ordering::Relaxed);
+        }
```

Comment:
Not sure how much it matters in practice, but shouldn't this be `Ordering::Release` which would imply a happens-before relationship for the writes in the critical section of `engine_state.repl_state`?

In `nu-cli/repl.rs` the `compare_exchange` which you set to `Acquire` is ahead of the read of the `repl_state` mutex which happens right before the call into `Reedline::read_line`.

I guess for singular uses of `commandline --accept` the mutex should ensure ordering in itself, as the `MutexGuard::drop` happens after the store.

Reply:
Good questions! I agree that the mutex guarantees overall ordering.

As for the use of `Relaxed`, for this specific flag, we don't care about other memory operations, because the flag itself is used in isolation and consumers of the flag are not dependent on other operations happening before this.

Same for the `compare_exchange`, which is used in isolation to determine whether we need to call the `line_editor.with_immediately_accept` API, which is also configured in isolation.